### PR TITLE
Improve error reporting for invalid refs

### LINKED ISSUE #36: Improve error handling for invalid refs
Passing an invalid ref currently fails by breaking everything without an error;
```checkedOption: refElement('input[type="radio"]:checked', { isRequired: false }),```

The first parameter should be the name of the ref, but instead a query-selector is passed. The proper way to do that would be; `refElement(parent => parent.querySelector('input[type="radio"]:checked'), ....)`

However, the component now silently crashes, so this should be handled better, with proper error reporting.

### PR DESCRIPTION
**Pull request by ThaNarie**
Output a proper warning that should educate the user what they are doing wrong.

fixes #36

### CODE REVIEW

**Review by psimk — APPROVED** [review]

File: src/lib/refs/refDefinitions.ts [hunk 1]
```diff
@@ -109,7 +109,19 @@ export function refElement<T extends RefElementType = RefElementType>(
       if (typeof refIdOrQuery === 'function') {
         element = refIdOrQuery(parent);
       } else {
-        element = (parent.querySelector<T>(`[data-ref="${this.ref}"]`) ?? null) as T | null;
+        try {
+          element = (parent.querySelector<T>(`[data-ref="${this.ref}"]`) ?? null) as T | null;
+        } catch (e) {
```
Comment (ThijsTyZ) [inline src/lib/refs/refDefinitions.ts:114]: could you change `e` to `error` to be more explicit?

File: src/lib/refs/refDefinitions.ts [hunk 2]
```diff
@@ -109,7 +109,19 @@ export function refElement<T extends RefElementType = RefElementType>(
       if (typeof refIdOrQuery === 'function') {
         element = refIdOrQuery(parent);
       } else {
-        element = (parent.querySelector<T>(`[data-ref="${this.ref}"]`) ?? null) as T | null;
+        try {
+          element = (parent.querySelector<T>(`[data-ref="${this.ref}"]`) ?? null) as T | null;
```
Comment (ThijsTyZ) [inline src/lib/refs/refDefinitions.ts:113]: How does this work? Usually `querySelector` doesn't throw an error if you try to select something that does not exist.

**Review by ThijsTyZ — CHANGES_REQUESTED** [review]

File: src/lib/refs/refDefinitions.ts (hunk 2 above)
Comment (psimk) [reply src/lib/refs/refDefinitions.ts:113]: The `try / catch` isn't to capture errors for elements that don't exist, but to capture errors for invalid queries.
See the related issue https://github.com/mubanjs/muban/issues/36, you can also try it in the browser console:
```js
document.querySelector('[data-ref="input[type="radio"]:checked"]')
```

Comment (ThijsTyZ) [reply src/lib/refs/refDefinitions.ts:113]: Got it. Thanks for the explanation

**Review by ThijsTyZ — APPROVED** [review]

**Pull request merged**